server/userModel/user.js

```diff
@@ -1,13 +1,196 @@
+ * @property {string} companyName - company name of the user.
+ * @property {string} password - password of the user.
+ */
+const User = mongoose.model('User', userSchema);
 
-module.exports = User;
\ No newline at end of file
+/**
+ * Given a user object, validate the user using it's model's schema, produce
+ * the new user's instance of a user and upload it to the database. If user
+ * object is missing either email, companyName or password, will return a usage
+ * string.
+ *
+ * @param {object} user - user object.
+ * @property {string} email - the user's email.
+ * @property {string} companyName - the user's company name.
+ * @property {string} password - the user's password.
+ * @returns {object|string}
+ */
+userSchema.statics.createAndUpload = async function ({
+  email,
+  companyName,
+  password,
+}) {
+  try {
+    await User.validate({ email, companyName }, ['email', 'companyName']);
+    const newUser = createNewUserObject({ email, companyName, password });
+    return saveDataToUserModel(newUser);
+  } catch (err) {
+    if (err instanceof mongoose.Error.ValidationError) {
+      const errorMessages = Object.keys(err.errors).map(
+        key => `Validation Error: ${key} needs to be present.`
+      );
+      return errorMessages.join('\n');
+    }
+  }
+};
+
+/**
+ * Produce an array of active users (not soft-deleted).
+ *
+ * @returns {object[]} an array of users.
+ * @static
+ */
+userSchema.statics.findAll = function () {
+  return this.find({ isActive: true });
+};
+
+/**
+ * Given an email, produce the user with the given email.
+ *
+ * @param {string} email - the email of the user to be found.
+ * @returns {object} - the user with the given email.
+ * @static
+ */
+userSchema.statics.findByEmail = function (email) {
+  return this.find({ email });
+};
+
+/**
+ * Given a company name, produce an array of users with the give company name.
+ *
+ * @param {string} companyName - the company name of the users to be found.
+ * @returns {object[]} - an array of users with the given company name.
+ * @static
+ */
+userSchema.statics.findByCompanyName = function (companyName) {
+  return this.find({ companyName });
+};
+
+/**
+ * Produce an array of all the users that have been (soft) deleted.
+ *
+ * @returns {object[]} - an array of users that have been (soft) deleted.
+ * @static
+ */
+userSchema.statics.findDeleted = function () {
+  return this.find({ isActive: false });
+};
+
+/**
+ * Produce true if the current user instance has been (soft) deleted. Else
+ * produce false.
+ *
+ * @returns {boolean} 'true' if the current the current user has been (soft)
+ * deleted. Else false.
+ * @method
+ */
+userSchema.methods.isDeleted = function () {
+  return !this.isActive;
+};
+
+/**
+ * Soft delete the current user instance.
+ *
+ * @returns {boolean|mongoose.Query} false if operation failed.
+ * @method
+ */
+userSchema.methods.softDelete = function () {
+  return User.updateOne(
+    { email: this.email },
+    { isActive: false },
```

Comment:
The user schema does not have isActive defined: you can get around needing to add this to every schema in the database by making a soft deletion plugin!

Reply:
Right! I forgot to add that. Thanks for catching it.

Cool, I didn't know that existed. I'll research how to do that.
Should I also add the property manually to the schema in the file?

Reply:
Resolved. Pushing now.